### ISSUE TITLE
No error on bind to non-existing exchange

### ISSUE THREAD

**Issue by fan-tom** [sozu-proxy/lapin#191]
I try to bind queue to non-existing exchange, but future is not resolved with error, instead I get such lines in log
```
[2019-08-06T12:26:34Z ERROR lapin::channel] Channel 3 closed by 50:20 => Soft(NOTFOUND) => NOT_FOUND - no exchange 'non_existing_exchange' in vhost '/'
[2019-08-06T12:26:34Z ERROR lapin::connection] Connection error
```
and future stays unresolved forever.
You may see failing test here https://github.com/fan-tom/lapin/blob/96e481b4e7953795a83ceb972af78c1536427ce7/futures/tests/connection.rs#L60

**Comment by Keruspe** [comment]
Thanks, will give it a look once I get back from vacations

**Comment by Keruspe** [comment]
Didn't properly dig into it yet, but could you try again with 0.26?

**Comment by fan-tom** [comment]
Unfortunately, no. Error handler set on client is invoked, but future is not resolved

**Comment by Keruspe** [comment]
Hmm, 0.26.1 might be better then. 0.26 fixed some connection error handling, 0.26.1 fixes the same wrt channel errors

**Comment by Keruspe** [comment]
Oh, this one is a little bit trickier than I thought. I'm fixing it now

**Comment by Keruspe** [comment]
This is now fixed in 0.26.3
